--- conf.py	2022-12-15 22:05:04.327634000 -0800
+++ conf.py.MOD	2022-12-15 21:55:18.301537000 -0800
@@ -82,7 +82,8 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = 'alabaster'
+#html_theme = 'alabaster'
+html_theme = 'sphinx_rtd_theme'
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
